### PR TITLE
[MTSRE-602] AMO: Validator for CSV deployments

### PR DESCRIPTION
**Pull request by Ankit152**
Signed-off-by: Ankit Kurmi <akurmi@redhat.com>

# AM0015

## Description

A new validator for CSV deployments:

* All deployments defined in the CSV must have resources requests and limits defined for both CPU and Memory.
* All deployments defined in the CSV must have readiness and liveness probe defined.
## Checklist

- [ ] Wiki page exists: https://github.com/mt-sre/addon-metadata-operator/wiki/AM0001
- [x] Followed the SOP: https://gitlab.cee.redhat.com/service/managed-tenants-sops/-/blob/main/MT-SRE/sops/AMO-validators.md
- [x] Tested against production addons in `managed-tenants/addons/`

### CODE REVIEW

File: pkg/validator/am0015/csv_deployments.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+			msg = append(msg, hasReadinessProbe.Reasons...)
+		}
+
+		if hasCPUresource := kube.HasCPUResourceRequirements(deployment); !hasCPUresource.Success {
+			msg = append(msg, hasCPUresource.Reasons...)
+		}
+
+		if hasMemory := kube.HasMemoryResourceRequirements(deployment); !hasMemory.Success {
+			msg = append(msg, hasMemory.Reasons...)
+		}
```
Comment (Ajpantuso) [inline pkg/validator/am0015/csv_deployments.go:84]: The linter should be used as an interface (See this [commit](https://github.com/Ajpantuso/addon-metadata-operator/commit/cd1214aa25acce7ce0ac7f57589f86b9255d669d))

File: pkg/validator/am0015/csv_deployments.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	}
+
+	if len(msg) > 0 {
+		return c.Fail(msg...)
+	}
+	return c.Success()
+}
+
+func getLatestBundle(bundles []*registry.Bundle) (*registry.Bundle, error) {
```
Comment (Ajpantuso) [inline pkg/validator/am0015/csv_deployments.go:93]: I would refactor this like so https://github.com/Ajpantuso/addon-metadata-operator/commit/985bca3c3bb3d534c775a739cadb10ec37c4d938

**Review by Ajpantuso — CHANGES_REQUESTED** [review]
Couple of comments, but otherwise this is looking good!

File: pkg/validator/am0015/csv_deployments.go (hunk 1 above)
Comment (Ankit152) [reply pkg/validator/am0015/csv_deployments.go]: Done! 🙂

File: pkg/validator/am0015/csv_deployments.go (hunk 2 above)
Comment (Ankit152) [reply pkg/validator/am0015/csv_deployments.go:81]: Fixed this in PR #152

**Review by Ajpantuso — APPROVED** [review]
Merging, but will disable on integration.

**Pull request merged**